runtimes/huggingface/tests/test_common.py

```diff
@@ -45,3 +47,82 @@ def test_load_pipeline(optimum_model: bool, expected):
     pipeline = load_pipeline_from_settings(hf_settings, model_settings)
 
     assert isinstance(pipeline.model, expected)
+
+
+@pytest.mark.parametrize(
+    "has_model_params, param_uri_value, expected",
+    [
+        (True, "", "./some-pretrained-folder"),
+        (True, None, "./some-pretrained-folder"),
+        (True, "/some/folder/model-artefacts", "./some-pretrained-folder"),
+        (False, "", "./some-pretrained-folder"),
+    ],
+)
+@patch("mlserver_huggingface.common._get_pipeline_class")
+def test_pipeline_was_initialised_and_pretrained_model_takes_precedence(
+    mock_pipeline_factory,
+    has_model_params: bool,
+    param_uri_value: Optional[str],
+    expected: str,
+):
+    mock_pipeline_factory.return_value = MagicMock()
+
+    hf_settings = HuggingFaceSettings(pretrained_model="./some-pretrained-folder")
+    model_params = None
+    if has_model_params:
+        model_params = ModelParameters(uri=param_uri_value)
+
+    model_settings = ModelSettings(
+        name="foo",
+        implementation=HuggingFaceRuntime,
+        parameters=model_params,
+    )
+
+    _ = load_pipeline_from_settings(hf_settings, model_settings)
+
+    mock_pipeline_factory.return_value.assert_called_once()
+    pipeline_call_args = mock_pipeline_factory.return_value.call_args
+
+    assert pipeline_call_args.kwargs["model"] == expected
+
+
+@pytest.mark.parametrize(
+    "empty_pretrained_model_value, has_model_params, param_uri_value, expected",
+    [
+        (None, True, None, None),
+        (None, True, "", ""),
+        (None, True, "/some/folder/model-artefacts", "/some/folder/model-artefacts"),
+        (None, False, "", None),
+        ("", True, None, None),
+        ("", True, "", ""),
+        ("", True, "/some/folder/model-artefacts", "/some/folder/model-artefacts"),
+        ("", False, "", ""),
+    ],
+)
+@patch("mlserver_huggingface.common._get_pipeline_class")
+def test_pipeline_was_initialised_when_pretrained_model_is_not_supplied(
```

Comment:
:confused: I'm confused by these tests for two reasons.  While the actual logic makes sense, the test names are verbose (what does "was initialised" mean in the context of the tests?) and the test logic is _identical_ between the two tests bar the value for `pretrained_model` as far as I can see.

Is this not clearer as a single, tabulated test?

Reply:
Good point about the test logic, I've refactored it to be more concise. About the test name: pipeline_is_initialised, I mean that when pipeline(name, model, ....) constructor is called, the correct model param is passed.